src/board.py

```diff
@@ -117,18 +96,18 @@ def __repr__(self) -> str:
             "\n▐\033[7m  A B C D E F G H  \033[0m▌\n\n"
         ).replace("None", " ")
 
-    def __setitem__(self, square: Square | Indices | None, piece: Piece | None):
+    def __setitem__(self, square: Square | str | None, piece: Piece | None):
         """Add a piece to a square.
 
         Args:
             square: The rank and file of the square.
             piece: The piece to be placed on the square.
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            self._board[i][j] = piece
+            square = Square(square)
+            self._board[square.rank][square.file] = piece
 
-    def __getitem__(self, square: Square | Indices | None) -> Piece | None:
+    def __getitem__(self, square: Square | str | None) -> Piece | None:
```

Comment:
Perhaps. We included in our algebra "orhpaned" pieces, not belonging on the board. Because `Board.square_of` might return `None`, this is necessary to close the algebra. That's the only such case really.